tiledb/sm/query/writer.cc

```diff
@@ -355,6 +414,55 @@ Status Writer::close_files(FragmentMetadata* meta) const {
   return Status::Ok();
 }
 
+Status Writer::compute_coord_dups(
+    const std::vector<uint64_t>& cell_pos,
+    std::set<uint64_t>* coord_dups) const {
+  auto coords_buff_it = attr_buffers_.find(constants::coords);
+  if (coords_buff_it == attr_buffers_.end())
+    return LOG_STATUS(
+        Status::WriterError("Cannot check for coordinate duplicates; "
+                            "Coordinates buffer not found"));
+
+  auto coords_buff = (unsigned char*)coords_buff_it->second.buffer_;
+  auto coords_size = array_schema_->coords_size();
+  auto coords_num = cell_pos.size();
+
+  for (uint64_t i = 1; i < coords_num; ++i) {
+    if (!memcmp(
+            coords_buff + cell_pos[i] * coords_size,
+            coords_buff + cell_pos[i - 1] * coords_size,
+            coords_size)) {
+      coord_dups->insert(cell_pos[i]);
+    }
+  }
+
+  return Status::Ok();
+}
+
+Status Writer::compute_coord_dups(std::set<uint64_t>* coord_dups) const {
+  auto coords_buff_it = attr_buffers_.find(constants::coords);
+  if (coords_buff_it == attr_buffers_.end())
+    return LOG_STATUS(
+        Status::WriterError("Cannot check for coordinate duplicates; "
+                            "Coordinates buffer not found"));
+
+  auto coords_buff = (unsigned char*)coords_buff_it->second.buffer_;
+  auto coords_buff_size = *coords_buff_it->second.buffer_size_;
+  auto coords_size = array_schema_->coords_size();
+  auto coords_num = coords_buff_size / coords_size;
+
+  for (uint64_t i = 1; i < coords_num; ++i) {
```

Comment:
We should profile this to see how expensive it is. If it's expensive we can do away with a separate `std::set<uint64_t>` and instead move to a model more like the read algorithm, which has an `OverlappingCoords` struct with a boolean `valid_` field. We can mark that field false in this function, and then this function will parallelize very well.